File: PC_Python/PC7_Notas.py
def leer_nota(mensaje):

    nota = -1

    while nota < 0 or nota > 20:
        nota = float(input(mensaje))

    return nota


def es_aprobado(nota):

    return nota >= 11


def clasificar_nota(nota):

    if nota >= 18:
        return "Excelente"

    elif nota >= 14:
        return "Bueno"

    elif nota >= 11:
        return "Aprobado"

    else:
        return "Desaprobado"


def calcular_promedio(suma, n):

    return suma / n


def mostrar_estadisticas(prom, mayor, menor, aprobados, total):

    print("Promedio:", prom)
    print("Mayor:", mayor)
    print("Menor:", menor)
    print("Aprobados:", aprobados)
    print("Total:", total)


suma = 0
aprobados = 0
mayor = 0
menor = 20

for i in range(3):

    nota = leer_nota("Ingrese nota: ")

    suma = suma + nota

    if es_aprobado(nota):
        aprobados = aprobados + 1

    if nota > mayor:
        mayor = nota

    if nota < menor:
        menor = nota

    print(clasificar_nota(nota))

prom = calcular_promedio(suma, 3)

mostrar_estadisticas(prom, mayor, menor, aprobados, 3)

File: PC_C_Char/PC7_Notas.cs
class Program
{
    static void Main()
    {
        double suma = 0;
        int aprobados = 0;
        double mayor = 0;
        double menor = 20;

        for (int i = 1; i <= 3; i++)
        {
            double nota = LeerNota("Ingrese nota:");

            suma = suma + nota;

            if (EsAprobado(nota))
            {
                aprobados++;
            }

            if (nota > mayor)
            {
                mayor = nota;
            }

            if (nota < menor)
            {
                menor = nota;
            }

            Console.WriteLine(ClasificarNota(nota));
        }

        double prom = CalcularPromedio(suma, 3);

        MostrarEstadisticas(prom, mayor, menor, aprobados, 3);

        Console.ReadKey();
    }

    static double LeerNota(string mensaje)
    {
        double nota;

        do
        {
            Console.WriteLine(mensaje);
            nota = Convert.ToDouble(Console.ReadLine());

        } while (nota < 0 || nota > 20);

        return nota;
    }

    static bool EsAprobado(double nota)
    {
        return nota >= 11;
    }

    static string ClasificarNota(double nota)
    {
        if (nota >= 18)
        {
            return "Excelente";
        }
        else if (nota >= 14)
        {
            return "Bueno";
        }
        else if (nota >= 11)
        {
            return "Aprobado";
        }
        else
        {
            return "Desaprobado";
        }
    }

    static double CalcularPromedio(double suma, int n)
    {
        return suma / n;
    }

    static void MostrarEstadisticas(double prom, double mayor, double menor, int aprobados, int total)
    {
        Console.WriteLine("Promedio: " + prom);
        Console.WriteLine("Mayor: " + mayor);
        Console.WriteLine("Menor: " + menor);
        Console.WriteLine("Aprobados: " + aprobados);
        Console.WriteLine("Total: " + total);
    }
}